src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/UsePropertiesWhereAppropriate.cs

```diff
@@ -76,8 +77,8 @@ public override void Initialize(AnalysisContext analysisContext)
                     if (methodSymbol.IsGenericMethod ||
                         methodSymbol.IsVirtual ||
                         methodSymbol.IsOverride ||
-                        methodSymbol.Name == GetHashCodeName ||
-                        methodSymbol.Name == GetEnumeratorName ||
+                        methodSymbol.Name is GetHashCodeName or GetEnumeratorName ||
+                        (methodSymbol.Name == GetPinnableReferenceName && (methodSymbol.ReturnsByRef || methodSymbol.ReturnsByRefReadonly)) ||
```

Comment:
Is there a better way to write this condition using the new pattern matching enhancement?

Reply:
It's non C# 9 but you could simply extract the code in a function `IsPinnableReferenceMethod`, also this could maybe be part of the helpers, I suspect some other rules will have to use it.

Reply:
Do we need to special case ref and ref readonly checks only to methods named `GetPinnableReference` or should we do it regardless of the method name? @sharwell for thoughts.

Reply:
Tag @sharwell again for the last question. Otherwise, LGTM.

Reply:
Going to keep the current restrictive check - if needed, we can adjust it in future.